File: cpp/src/clc.cpp
// Copyright © 2022 Mark Summerfield. All rights reserved.
// License: GPLv3

#include <iostream>
#include "config.hpp"

using namespace std;


int main() {
    cout << "clc C++\n";
    Config config;
    try {
        config.update_from_command_line();
    } catch (exception& err) {
        cerr << "error: " << err.what() << '\n';
        return 1;
    }
}

File: go/clc.go
// Copyright © 2022 Mark Summerfield. All rights reserved.
// License: Apache-2.0

package main

import (
	"time"
)

func main() {
	config := getConfig()
	t := time.Now()
	files := getFilenames(&config)
	fileData := processFiles(files, &config)
	if config.summary {
		displaySummary(config.dataForLang, fileData, config.sortByLines,
			time.Since(t))
	} else {
		displayFull(config.dataForLang, fileData, config.sortByLines,
			config.maxWidth)
	}
}
